packages/data-store/src/digitalCredential/DigitalCredentialStore.ts

```diff
@@ -0,0 +1,116 @@
+    debug('Adding credential', args)
+    const digitalCredentialEntityRepository: Repository<DigitalCredentialEntity> = (await this.dbConnection).getRepository(DigitalCredentialEntity)
+    const credentialEntity: NonPersistedDigitalCredential = nonPersistedDigitalCredentialEntityFromAddArgs(args)
+    const createdResult: DigitalCredentialEntity = await digitalCredentialEntityRepository.save(credentialEntity)
+    return Promise.resolve(createdResult)
+  }
+
+  getDigitalCredential = async (args: GetDigitalCredentialArgs): Promise<DigitalCredentialEntity> => {
+    const result: DigitalCredentialEntity | null = await (await this.dbConnection).getRepository(DigitalCredentialEntity).findOne({
+      where: args,
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No credential found for arg: ${args.toString()}`))
+    }
+    return result
+  }
+
+  getDigitalCredentials = async (args?: GetDigitalCredentialsArgs): Promise<GetDigitalCredentialsResponse> => {
+    const { filter = {}, skip, take, order = { createdAt: 'DESC' } } = args ?? {}
+    const [result, total] = await (await this.dbConnection).getRepository(DigitalCredentialEntity).findAndCount({
+      where: filter,
+      skip,
+      take,
+      order,
+    })
+    if (!result) {
+      return Promise.reject(Error(`No credential found for arg: ${args?.toString()}`))
+    }
+    return {
+      data: result,
+      total,
+      hasMore: total > (skip || 0) + (take || 0),
+    }
+  }
+
+  removeDigitalCredential = async (args: RemoveDigitalCredentialArgs): Promise<boolean> => {
+    if (!args) {
+      return false
+    }
+
+    let query: FindOptionsWhere<DigitalCredentialEntity> = {}
+
+    if ('id' in args) {
+      query.id = args.id
+    } else if ('hash' in args) {
+      query.hash = args.hash
+    } else {
+      return false
+    }
+    try {
+      const connection = await this.dbConnection
+      const result = await connection.getRepository(DigitalCredentialEntity).delete(query)
+      return result.affected === 1
+    } catch (error) {
+      console.error('Error removing digital credential:', error)
+      return false
+    }
+  }
+
+  updateDigitalCredentialState = async (args: UpdateDigitalCredentialStateArgs): Promise<DigitalCredentialEntity> => {
+    const credentialRepository: Repository<DigitalCredentialEntity> = (await this.dbConnection).getRepository(DigitalCredentialEntity)
+    const whereClause: Record<string, any> = {}
+    if ('id' in args) {
+      whereClause.id = args.id
+    }
+    if ('hash' in args) {
+      whereClause.hash = args.hash
+    }
+    const credential: DigitalCredentialEntity | null = await credentialRepository.findOne({
+      where: whereClause,
+    })
+
+    if (!credential) {
+      return Promise.reject(Error(`No credential found for args: ${whereClause}`))
+    }
+    const updatedCredential: DigitalCredential = {
+      ...credential,
+      lastVerificationDate: args.verificationDate ?? new Date(),
```

Comment:
not sure if I understand this comment. this is updating the state of a stored credential and not adding a new credential

Reply:
The adding part wasn't the point. Read 'updating'. **"automatically result in a lastVerificationDate if not provided"** The next sentence should also be pretty clear what the issue is with the way you are setting the lastVerificationDate.
I suggest you reread the remarks, read the code setting the lastVerificationDate and then tell me in your own words, what is wrong with it, because it should be immediately clear to you if you ask me

Reply:
if I understand correctly, we should always get a date in the params of the `updateCredentialState` it's `revokedAt` for when it's revoked and `verifiedAt` when it's to another state. and never should set it with current date.

Reply:
Yes, but also only set or overwrite it if the verification date is passed in. It is optional, because I could just as well be storing a VC/VP without verifying, or just playnly storing a C/P

Reply:
The most important thing of course was that persisting a (V)C/(V)P and providing some properties to make it more easily accessible for indexing/UX etc, has 0 relation with the store determining by itself when the last verification apparently happened if it was not passed in as argument

Reply:
I hope I got it right this time. no setting the `verifiedAt` date when it's not provided. `verifiedAt` is optional in the entity itself.